actors/common/src/main/java/org/sunbird/learner/actors/data/security/manager/DataSecurityBackgroundActor.java

```diff
@@ -0,0 +1,185 @@
+                && ((String) userMap.get(JsonKey.PHONE)).length() <= maxPhoneNoSizeAllowed)) {
+          encryptUserDataAndUpdateDb(userMap);
+          userIdsListToSync.add((String) userMap.get(JsonKey.ID));
+          i++;
+        }
+      } else {
+        if ((StringUtils.isNotBlank((String) userMap.get(JsonKey.EMAIL))
+                && !(ProjectUtil.isEmailvalid((String) userMap.get(JsonKey.EMAIL)))
+            || (StringUtils.isNotBlank((String) userMap.get(JsonKey.PHONE))
+                && ((String) userMap.get(JsonKey.PHONE)).length() > maxPhoneNoSizeAllowed))) {
+          decryptUserDataAndUpdateDb(userMap);
+          userIdsListToSync.add((String) userMap.get(JsonKey.ID));
+          i++;
+        }
+      }
+    }
+    ProjectLogger.log(
+        "EsSyncBackgroundActor:commonEncryptionDecryption: total number of user details"
+            + (isEncrypted ? " encrypted " : " decrypted ")
+            + "is: "
+            + i,
+        LoggerEnum.INFO);
+    if (CollectionUtils.isNotEmpty(userIdsListToSync)) {
+      syncToES(userIdsListToSync);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private List<Map<String, Object>> getUserDetails(Request request) {
+    List<String> userIds = (List<String>) request.getRequest().get(JsonKey.USER_IDs);
+    Response response =
+        cassandraOperation.getRecordsByIdsWithSpecifiedColumns(
+            JsonKey.SUNBIRD, JsonKey.USER, null, userIds);
+    List<Map<String, Object>> userList = (List<Map<String, Object>>) response.get(JsonKey.RESPONSE);
+    if (CollectionUtils.isEmpty(userList)) {
+      ProjectCommonException.throwClientErrorException(ResponseCode.invalidUserId);
+    }
+    return userList;
+  }
+
+  private void decryptUserDataAndUpdateDb(Map<String, Object> userMap) {
+    try {
+      UserUtility.decryptUserData(userMap);
+      cassandraOperation.updateRecord(usrDbInfo.getKeySpace(), usrDbInfo.getTableName(), userMap);
+      getUserAddressDataDecryptAndUpdateDb((String) userMap.get(JsonKey.ID));
+    } catch (Exception e) {
+      ProjectLogger.log(
+          "DataSecurityBackgroundActor:decryptUserDataAndUpdateDb: Exception Occurred while decrypting user data for userId "
+              + ((String) userMap.get(JsonKey.ID)),
+          e);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private void getUserAddressDataDecryptAndUpdateDb(String userId) {
+    Response response =
+        cassandraOperation.getRecordsByProperty(
+            addrDbInfo.getKeySpace(), addrDbInfo.getTableName(), JsonKey.USER_ID, userId);
+    List<Map<String, Object>> addressList =
+        (List<Map<String, Object>>) response.get(JsonKey.RESPONSE);
+    try {
+      UserUtility.decryptUserAddressData(addressList);
+      for (Map<String, Object> address : addressList) {
+        cassandraOperation.updateRecord(
+            addrDbInfo.getKeySpace(), addrDbInfo.getTableName(), address);
+      }
+    } catch (Exception e) {
+      ProjectLogger.log(
```

Comment:
catch specific exception

Reply:
UserUtility.decryptUserAddressData(addressList) -> this method throws Exception type.